### DISCUSSION TITLE
Search function error - unhandled runtime error

### DISCUSSION THREAD

**Discussion by dramaticgarlic** [neuefische/questions#660] [Web] [answered]
Hello everyone!

I am working on my search function and when trying to filter for the tags property in my object dummy data I am getting the error below. I tried using .test instead of .some and the error appears as well and in that case I also get an error. I am definitely missing something but for now I need an extra pair of fresh eyes who can maybe spot the issue. 

## current error 

![Screenshot 2023-08-04 at 10 50 27](https://github.com/IC-Socaciu/all-access-pass-1/assets/125737167/bb2947c6-ea9c-489b-829b-7c4e8cdfb086)

 ```Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'some')

Source
public/interviews.js (349:19) @ some

  347 | export default function SearchResults({ searchTerm }) {
  348 | const results = interviews.filter((interview) =>
> 349 |   interview.tags.some((tag) => tag.includes(searchTerm))
      |                 ^
  350 | );
  351 | 
  352 | if (!searchTerm || !results.length) {
```

 ## PR Link
https://github.com/IC-Socaciu/all-access-pass-1/pull/13

Any help is highly appreciated!

**Comment by Kheirah** [comment, marked as answer]
Your data is not consistent. Only the first entry has a `tags` attribute while the other entries don't. That's the cause for the above error.

**Comment by dramaticgarlic** [comment]
Thank you! that solved part of it indeed! And I had to add the SearchResults as a component as well.
